### PR TITLE
rippled: 1.4.0 -> 1.5.0

### PR DESCRIPTION
**Pull request by RaghavSood**
###### Motivation for this change

Ripple has undergone a network upgrade and rippled 1.4.0 no longer syncs

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6486,6 +6486,12 @@
     githubId = 131856;
     name = "Arnout Engelen";
   };
+  RaghavSood = {
```
Comment (marsam) [inline maintainers/maintainer-list.nix:6489]: please separate this into another commit

File: pkgs/servers/rippled/default.nix [hunk 2]
```diff
@@ -60,84 +82,71 @@ let
...
+    fetchSubmodules = true;
   };
 
   hardeningDisable = ["format"];
-  cmakeFlags = [
-    "-Dstatic=OFF"
-    "-DBOOST_LIBRARYDIR=${boost.out}/lib"
-    "-DBOOST_INCLUDEDIR=${boost.dev}/include"
-  ];
+  cmakeFlags = ["-Dstatic=OFF" "-DBoost_NO_BOOST_CMAKE=ON" "-DCMAKE_BUILD_TYPE=Release"];
```
Comment (marsam) [inline pkgs/servers/rippled/default.nix:130]: cmake hook already adds `CMAKE_BUILD_TYPE`
```suggestion
  cmakeFlags = ["-Dstatic=OFF" "-DBoost_NO_BOOST_CMAKE=ON" ];
```

File: pkgs/servers/rippled/default.nix [hunk 3]
```diff
@@ -60,84 +82,71 @@ let
...
 
   nativeBuildInputs = [ pkgconfig cmake git ];
-  buildInputs = [ openssl openssl.dev zlib ];
+  buildInputs = [ openssl openssl.dev boostSharedStatic zlib grpc c-ares c-ares.cmake-config abseil-cpp protobuf3_8 ];
 
   preConfigure = ''
     export HOME=$PWD
 
+    echo $HOME && ls -al
```
Comment (marsam) [inline pkgs/servers/rippled/default.nix:138]: would you mind deleting this line?

Comment (RaghavSood) [reply pkgs/servers/rippled/default.nix]: Apologies, that was for debugging - removed

File: pkgs/servers/rippled/default.nix (hunk 2 above)
Comment (RaghavSood) [reply pkgs/servers/rippled/default.nix]: Ah, interesting - updated

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (RaghavSood) [reply maintainers/maintainer-list.nix:6489]: Done

**Review by offlinehacker — APPROVED** [review]
Thank you for update, looks good 🙂

**Comment by offlinehacker** [comment]
We used rippled from nixpkgs in my previous company, but since I left, I did not keep it updated, well they seem to not use this package anymore.

**Comment by RaghavSood** [comment]
@offlinehacker thanks for the merge! I work at an exchange, and we do use rippled :) Should be able to keep it up to date with minimal effort
